src/navigator/Navigator.tsx

```diff
@@ -521,6 +528,17 @@ const generalScreens = (Navigator: typeof Stack) => (
       component={ProfileMenu}
       options={ProfileMenu.navigationOptions as NativeStackNavigationOptions}
     />
+    <Navigator.Screen
+      name={Screens.Settings}
+      component={SettingsScreen}
+      options={isTabNav ? headerWithBackButton : noHeader}
```

Comment:
could we just register the back button always? I think this version of the screen should only be used in the tab navigator anyway. The screens I believe are registered at startup and won't be refreshed if the gate changes when app is backgrounded.